### PR TITLE
[1.4.x] Fix bug in resource with catch-all path

### PR DESCRIPTION
**Pull request by aneeshafedo**
## Purpose
$subject
> Resolves https://github.com/wso2-enterprise/internal-support-ballerina/issues/301

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: openapi-core/src/main/java/io/ballerina/openapi/core/generators/service/BallerinaServiceGenerator.java [hunk 1]
```diff
@@ -268,7 +269,8 @@ private FunctionDefinitionNode getResourceFunction(Map.Entry<PathItem.HttpMethod
             isNullableRequired = parametersGenerator.isNullableRequired();
         }
         SeparatedNodeList<ParameterNode> parameters = createSeparatedNodeList(params);
-        String pathForRecord = Objects.equals(path, SLASH) ? "" : GeneratorUtils.getValidName(path, true);
+        String pathForRecord = Objects.equals(path, SLASH) || Objects.equals(path, CATCH_ALL_PATH) ? "" :
+                GeneratorUtils.getValidName(path, true);
```
Comment (NipunaRanasinghe) [inline openapi-core/src/main/java/io/ballerina/openapi/core/generators/service/BallerinaServiceGenerator.java:273]: Any specific reason to use `Objects.equals()` in here? Can't we just do `str1.equals(str2)` instead?

**Review by NipunaRanasinghe — APPROVED** [review]
LGTM

File: openapi-cli/src/test/resources/petstore_catch_all_path.yaml [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+      flows:
+        implicit:
+          authorizationUrl: https://petstore3.swagger.io/oauth/authorize
+          scopes:
+            write:pets: modify pets in your account
+            read:pets: read your pets
+    api_key:
+      type: apiKey
+      name: api_key
+      in: header
```
Comment (lnash94) [inline openapi-cli/src/test/resources/petstore_catch_all_path.yaml:213]: Missing new line

**Review by lnash94 — APPROVED** [review]
Approve with a minor suggestion

**Pull request merged**